apps/advisor/pkg/app/checks/datasource/check.go

```diff
@@ -0,0 +1,90 @@
+package datasource
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	advisor "github.com/grafana/grafana/apps/advisor/pkg/apis/advisor/v0alpha1"
+	"github.com/grafana/grafana/apps/advisor/pkg/app/checks"
+	"github.com/grafana/grafana/pkg/apimachinery/identity"
+	"github.com/grafana/grafana/pkg/services/datasources"
+	"github.com/grafana/grafana/pkg/services/user"
+	"github.com/grafana/grafana/pkg/util"
+	"k8s.io/klog/v2"
+)
+
+func init() {
+	checks.AddFactory(&factory{})
+}
```

Comment:
Will we always have to register checks this way?

Reply:
I tend to try avoid `init` and global state (though unexported) can be a bit painful too. It's probably fine in the short term, but maybe we just have different registries for OSS + Enterprise that are setup with Wire?

Reply:
I'll have a quick play around and see if the idea I have fits

Reply:
Rough idea here: https://github.com/grafana/grafana/compare/advisorAppDSCheck...advisor-registry?expand=1

I think `advisorConfig` the way it is will just get bigger and bigger as it will need to take on more dependencies, so I figured to rework that into it too. The main idea is:
- Keep checks simple so people can write them easily like the one any other Grafana service (could easily just covert to Wire services if necessary)
- Removes the need for duplicated factory code for each check
- Allows to easily have a enterprise specific registry implementation with enterprise only checks

PS: `SpecificConfig` should probably still remain a struct so it can have other things as well as the registry.

Reply:
got it! It always takes me some time to wrap my head around wire injections :) I checked what would be needed to have an enterprise-only check and found it doable (even though requires some changes that I won't do here for simplicity, we can do that when needed). Thanks for the suggestion!